components/sysutils/p0f/patches/001-p0f-3.09b-solarisfix.patch

```diff
@@ -0,0 +1,859 @@
++    ### p0f-daemon service is at all enabled (optional_all), otherwise this
++    ### particular client starts up unimpeded without p0f (and fails to query
++    ### the fingerprinter, but otherwise just works).
++-->
++<!--
++    <dependent name='milter-greylist' grouping='optional_all' restart_on='none'>
++      <service_fmri value='svc:/network/milter-greylist:default'/>
++    </dependent>
++-->
++    <exec_method name='start' type='method' exec='/usr/sbin/p0f.sh %m' timeout_seconds='120'>
++      <method_context>
++            <method_credential
++            user='root'
++            group='root'
++            limit_privileges=':default'
++	    privileges='basic,net_icmpaccess,net_rawaccess,file_chown,file_owner,file_dac_write,proc_setid,proc_chroot'/>
++<!--   ### The set of privileges above allows the SMF-ized daemon to access
++       ### the low-level networking. Normally SMF services are prohibited
++       ### from such intimacy with the OS in order to improve security.
++       ### Note we start as "root" above and later drop privileges in script. -->
++      </method_context>
++    </exec_method>
++    <exec_method name='stop' type='method' exec=':kill' timeout_seconds='60'>
++      <method_context/>
++    </exec_method>
++    <exec_method name='refresh' type='method' exec=':kill' timeout_seconds='180'>
++      <method_context/>
++    </exec_method>
++    <property_group name='general' type='framework'>
++      <propval name='action_authorization' type='astring' value='solaris.smf.manage.milter'/>
++    </property_group>
++    <property_group name='startd' type='framework'>
++      <propval name='ignore_error' type='astring' value='core,signal'/>
++    </property_group>
++    <stability value='Unstable'/>
++    <template>
++      <common_name>
++        <loctext xml:lang='C'>p0f passive fingerprinter service</loctext>
++      </common_name>
++      <documentation>
++        <manpage title='p0f' section='8' manpath='/usr/share/man'/>
++      </documentation>
++    </template>
++  </service>
++</service_bundle>
+diff --git p0f-3.09b-orig/p0f.c p0f-3.09b-jimfork/p0f.c
+index 7f17587..048b682 100644
+--- p0f-3.09b-orig/p0f.c	2014-11-08 06:55:17.000000000 +0300
++++ p0f-3.09b-jimfork/p0f.c	2015-06-18 03:47:00.573916124 +0300
+@@ -15,7 +15,9 @@
+ #include <stdlib.h>
+ #include <unistd.h>
+ #include <signal.h>
+-#include <getopt.h>
++#ifndef DONT_HAVE_GETOPT_H
++#  include <getopt.h>
++#endif
+ #include <errno.h>
+ #include <dirent.h>
+ #include <pwd.h>
+@@ -28,12 +30,31 @@
+ #include <sys/socket.h>
+ #include <arpa/inet.h>
+ #include <sys/un.h>
+-#include <sys/fcntl.h>
++#ifdef SOLARIS
++//#  warning "SOLARIS=yes"
++#  include <sys/fcntl.h>
++#  include <fcntl.h>
++#  include <sys/ioctl.h>
++#  include <sys/stream.h>
++#  include <sys/dlpi.h>
++#  include <sys/bufmod.h>
++#  include <sys/stropts.h>
++# ifdef SOLARIS_UCB
++#    include "/usr/ucbinclude/sys/file.h"
```

Comment:
I would prefer not to, because it would make maintenance of the patch harder for me (currently I can `git diff master` in a checkout of https://github.com/jimklimov/p0f @ `master-solaris`). Anyhow, the UCB code is by default disabled (in `build.sh`)...